streaming/streaming.go

```diff
@@ -0,0 +1,78 @@
+package streaming
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	"github.com/sul-dlss-labs/taco/config"
+)
+
+// Stream the interface for streaming pipeline
+type Stream interface {
+	SendMessage(string) error
+	// Do we want to expose these methods?
+	// createStream()
+	// waitUntilStreamExists()
+}
+
+type kinesisStream struct {
+	streamName *string
+	connection *kinesis.Kinesis
+}
+
+// NewKinesisStream create a new kinesis stream
+func NewKinesisStream(config *config.Config) (Stream, error) {
+	s := session.New(&aws.Config{Region: aws.String(config.AWSRegion),
+		Endpoint:   aws.String(config.KinesisEndpoint),
+		DisableSSL: aws.Bool(config.KinesisDisableSSL)})
+
+	kc := kinesis.New(s)
+
+	streamName := aws.String(config.DepositStreamName)
+
+	return &kinesisStream{streamName: streamName, connection: kc}, nil
+}
+
+func (d kinesisStream) SendMessage(message string) error {
+	// d.createStream()
+	// d.waitUntilStreamExists()
+
```

Comment:
Are these 2 comments related to your question above or should they be removed?

Reply:
This was some of the capabilities we have, but are not currently exercising.